docs/querying/set-query-context.md

```diff
@@ -0,0 +1,240 @@
+  -->
+  
+
+The query context gives you fine-grained control over how Apache Druid executes your individual queries. While the default settings in Druid work well for most queries, you can set query context to handle specific requirements and optimize performance.
+
+Common use cases for the query context include:
+- Override default timeouts for long-running queries or complex aggregations.
+- Control resource usage to prevent expensive queries from overwhelming your cluster.
+- Debug query performance by disabling caching during testing.
+- Configure SQL-specific behaviors like time zones for accurate time-based analysis.
+- Set priorities to ensure critical queries get computational resources first.
+- Adjust memory limits for queries that process large datasets.
+
+Druid provides several ways to set query context, and the method you use depends on how and where you're submitting your query.
+This guide lists how to set query context for each method of submitting queries.
+
+Before you begin, identify which context parameters you need to configure in order to establish your query context as query context carriers. For available parameters and their descriptions, see [Query context reference](query-context-reference.md).
+
+## Druid web console
+
+The most straightforward method to configure query context parameters is via the Druid web console. In web console, you can set up context parameters for both Druid SQL and native queries.
+
+The following steps outline how to define query context parameters:
+
+1. Open the **Query** tab in the web console.
+
+1. **Click** the **Engine** selector next to the **Run** button to choose the appropriate query type:
+
+- Selects the **JSON (native) engine** for native queries.
+- Selects the **SQL (native) engine** for Druid SQL queries.
+- Selects the **SQL (task) engine** for Multi-stage queries (MSQ).
+- Selects the **Auto engine** to let the console detect the query type automatically. You just paste your query into the **Query** view, and the web console chooses the right engine for you.
+
+2. Enter the query you want to run in the web console.
+
+3. Select **Edit context** from the menu.
+4. In the **Edit query context** dialog, add your context parameters as JSON key-value pairs:
+   ```json
+   {
+     "timeout": 300000,
+     "useCache": false
+   }
+   ```
+5. Click **Save** to apply the context to your query.
+6. Click **Run** to execute your query with the specified context parameters.
+
+The web console validates your JSON and highlights any syntax errors before you run the query.
+
+For more information about using the Druid SQL Web Console Query view, see [Query view](../operations/web-console.md#query).
+
+## Druid SQL
+
+When you use Druid SQL programmatically—such as in applications, automated scripts, or database tools, you can set query context using three different methods depending on how you execute your queries beyond [Druid Web Console](./set-query-context.md#druid-web-console). 
```

Comment:
Druid Web Console -- fix casing. Also check how this is referred to in other places. Is it called "Druid web console" everywhere or just "web console"?